pilot/pkg/xds/xdsgen.go

```diff
@@ -92,8 +92,17 @@ func (s *DiscoveryServer) pushXds(con *Connection, w *model.WatchedResource, req
 	if w == nil {
 		return nil
 	}
+
+	if features.FilterGatewayClusterConfig && w.TypeUrl == v3.RouteType {
+		for !con.gotAckForLastResponse(v3.ClusterType) {
+			log.Infof("Last sent cds not acked yet. Hold rds push")
+			time.Sleep(time.Millisecond * 500)
```

Comment:
not sure on this duration that it is too long or too short.

Reply:
It's not, any sleep like this is not going to be acceptable IMO.